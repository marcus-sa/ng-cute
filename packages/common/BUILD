load("@npm_bazel_typescript//:index.bzl", "ts_library")
load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package")

package(default_visibility = ["//visibility:public"])

ts_library(
  name = "common",
  module_name = "@ng-qt/common",
  srcs = glob(
    include = ["**/*.ts"],
    exclude = ["**/*.spec.ts"],
  ),
  deps = [
    "@npm//@angular/core",
    "@npm//@nodegui/nodegui",
    "@npm//change-case",
    "@npm//tslib",
  ],
)

npm_package(
  name = "npm",
  deps = [
    ":common",
  ],
  srcs = [
    "package.json",
  ],
)