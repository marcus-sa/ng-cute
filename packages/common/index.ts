export * from './src/interfaces';
export * from './src/widget.decorator';
export * from './src/nodes';

export { APP_ROOT_VIEW } from './src/tokens';
export {
  getWidgetMeta,
  isNodeWidget,
  isFlexLayout,
  isNodeLayout,
  // isParentNodeFlexLayout,
  isDetachedElement,
  isFunc,
  isInstance,
  isInvisibleNode,
  isNil,
  isStr,
  hasViewMeta,
  throwIfAlreadyLoaded,
  isView,
} from './src/utils';
