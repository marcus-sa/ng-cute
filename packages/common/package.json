{
  "name": "@ng-qt/common",
  "version": "0.0.1",
  "main": "src/index.js",
  "dependencies": {
    "change-case": "*"
  },
  "peerDependencies": {
    "@nodegui/nodegui": "*"
  }
}
