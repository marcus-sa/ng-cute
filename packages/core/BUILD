load("@npm_bazel_typescript//:index.bzl", "ts_library")
load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package")
load("@npm_angular_bazel//:index.bzl", "ng_module")

package(default_visibility = ["//visibility:public"])

ng_module(
    name = "core",
    srcs = glob(
        include = ["**/*.ts"],
        exclude = ["**/*.spec.ts"],
    ),
    module_name = "@ng-qt/core",
    deps = [
        "//packages/common",
        "//packages/platform",
        "@npm//@angular/common",
        "@npm//@angular/core",
        "@npm//@angular/platform-browser",
        "@npm//@nodegui/nodegui",
        "@npm//@types",
        "@npm//rxjs",
        "@npm//tslib",
    ],
)

npm_package(
    name = "npm",
    srcs = [
        "package.json",
    ],
    deps = [
        ":core",
    ],
)
