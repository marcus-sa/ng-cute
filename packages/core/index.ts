export * from './src/ng-qt.module';
