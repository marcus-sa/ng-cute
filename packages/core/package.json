{
  "name": "@ng-qt/core",
  "version": "0.0.0-PLACEHOLDER",
  "dependencies": {
    "@angular/platform-browser": "8.2.12",
    "@ng-qt/platform": "0.0.0-PLACEHOLDER",
    "@ng-qt/common": "0.0.0-PLACEHOLDER",
    "tslib": "1.10.0"
  },
  "peerDependencies": {
    "@angular/core": "8.2.12",
    "@nodegui/nodegui": "0.2.1",
    "rxjs": "6.5.3",
    "zone.js": "0.10.2"
  },
  "sideEffects": false
}
