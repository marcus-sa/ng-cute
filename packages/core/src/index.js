'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
const tslib_1 = require('tslib');
tslib_1.__exportStar(require('./ng-qt.module'), exports);
tslib_1.__exportStar(require('./ng-qt-widget'), exports);
tslib_1.__exportStar(require('./register-widget'), exports);
//# sourceMappingURL=index.js.map
