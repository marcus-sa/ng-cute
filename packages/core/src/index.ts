export * from './ng-qt.module';
export * from './ng-qt-widget';
export * from './utils';
