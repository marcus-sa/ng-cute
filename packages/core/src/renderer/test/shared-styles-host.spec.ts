import { NgQtView } from '@ng-qt/common';
import { NgZone } from '@angular/core';

import {
  CONTENT_ATTR,
  HOST_ATTR,
  InlineStyles,
  NgQtSharedStylesHost,
} from '../shared-styles-host';

const testInlineStyle = 'color:red;font-size:12px';

describe('NgQtSharedStylesHost', () => {
  let sharedStylesHost: any | NgQtSharedStylesHost;
  let ngZone: any | NgZone;
  let node: any | NgQtView = {};

  beforeEach(() => {
    ngZone = new NgZone({});
    sharedStylesHost = new NgQtSharedStylesHost(ngZone);

    node.setStyleSheet = jest.fn();
    node.styles = new Map([['color', 'red'], ['font-size', '12px']]);

    sharedStylesHost.setNodeInlineStyle = jest.fn();
    // invoke tasks immediately
    ngZone.run = (cb: Function) => cb();
  });

  describe('nodeStylesToInlineStyle', () => {
    it(`should return "${testInlineStyle}"`, () => {
      const inlineStyles = sharedStylesHost.nodeStylesToInlineStyle(node);
      expect(inlineStyles).toMatchSnapshot();
      expect(inlineStyles).toEqual(testInlineStyle);
    });
  });

  describe('removeInlineStyle', () => {
    it('should remove property from node styles', () => {
      sharedStylesHost.removeInlineStyle(node, 'font-size');

      expect(node.styles).toMatchSnapshot();
      expect(node.styles).toEqual(new Map([['color', 'red']]));
      // expect(sharedStylesHost.setNodeInlineStyle).toHaveBeenCalledWith(node);
    });
  });

  describe('addInlineStyle', () => {
    it('should add style property if inline styles is object', () => {
      const inlineStyles: InlineStyles = {
        property: 'max-width',
        value: '50px',
      };

      sharedStylesHost.addInlineStyle(node, inlineStyles);

      expect(node.styles).toMatchSnapshot();
      expect(node.styles).toMatchObject(
        new Map([...node.styles, ['max-width', '50px']]),
      );
    });

    it('should convert inline styles to node styles', () => {
      sharedStylesHost.addInlineStyle(node, testInlineStyle);

      expect(node.styles).toMatchSnapshot();
      expect(node.styles).toMatchObject(
        new Map([['color', 'red'], ['font-size', '12px']]),
      );
    });
  });

  describe('addHostStyles', () => {
    let type: any;

    beforeEach(() => {
      type = {};
    });

    it('should not set stylesheet if styles are empty', () => {
      type.styles = ['  ', '', '\n', '\r\n'];

      sharedStylesHost.addHostStyles(node, type);
      expect(node.setStyleSheet).not.toHaveBeenCalled();
    });

    it(`should replace [_nghost-%COMP%] with component id`, () => {
      type.id = 'c0';
      type.styles = [
        `
        ${HOST_ATTR} {
          padding: 24px;
        }
      `,
      ];

      sharedStylesHost.addHostStyles(node, type);
      expect(node.setStyleSheet).toHaveBeenCalledWith(
        `
        #${type.id} {
          padding: 24px;
        }
      `.trim(),
      );
    });

    // `should remove ${CONTENT_ATTR} from styles`
    it(`should remove [_ngcontent-%COMP%] from styles`, () => {
      type.styles = [
        `
        #btn${CONTENT_ATTR} {
          color: blue;
        }
         
        #danger-btn${CONTENT_ATTR} {
          color: red;
        }
      `,
      ];

      sharedStylesHost.addHostStyles(node, type);
      expect(node.setStyleSheet).toHaveBeenCalledWith(
        `
        #btn {
          color: blue;
        }
         
        #danger-btn {
          color: red;
        }
      `.trim(),
      );
    });
  });
});
