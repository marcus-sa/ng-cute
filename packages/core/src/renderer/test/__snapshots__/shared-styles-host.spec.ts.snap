// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgQtSharedStylesHost addInlineStyle should add style property if inline styles is object 1`] = `
Map {
  "color" => "red",
  "font-size" => "12px",
  "max-width" => "50px",
}
`;

exports[`NgQtSharedStylesHost addInlineStyle should convert inline styles to node styles 1`] = `
Map {
  "color" => "red",
  "font-size" => "12px",
}
`;

exports[`NgQtSharedStylesHost nodeStylesToInlineStyle should return "color:red;font-size:12px" 1`] = `"color:red;font-size:12px"`;

exports[`NgQtSharedStylesHost removeInlineStyle should remove property from node styles 1`] = `
Map {
  "color" => "red",
}
`;
